pyccel/codegen/printing/ccode.py

```diff
@@ -1183,25 +1205,37 @@ def _print_OMP_Single_Construct(self, expr):
     def _print_Omp_End_Clause(self, expr):
         return '}'
     #=====================================
+    def free_allocs(self, variables):
+        allocs = set()
+        for var in variables:
+            for v in var:
+                allocs.add(v)
+        if len(allocs) > 0:
+            allocs_str = ', '.join(self._print(i.name) for i in allocs)
+            self._allocs = self._allocs[:len(self._allocs) - 2]
```

Comment:
Why do we remove the last 2 elements?

In python negative indexes are allowed and are translated to the line you have written. This line would normally be written as: `self._allocs = self._allocs[:-2]`